beacon-chain/core/blocks/block_operations.go

```diff
@@ -774,6 +774,29 @@ func ProcessDeposits(ctx context.Context, beaconState *pb.BeaconState, body *eth
 	return beaconState, nil
 }
 
+// ProcessPreGenesisDeposit processes a deposit for the beacon state before chainstart.
+func ProcessPreGenesisDeposit(ctx context.Context, beaconState *pb.BeaconState,
+	deposit *ethpb.Deposit, valIndexMap map[[48]byte]int) (*pb.BeaconState, error) {
+	var err error
+	beaconState, err = ProcessDeposit(beaconState, deposit, valIndexMap)
+	if err != nil {
+		return nil, errors.Wrap(err, "could not process deposit")
+	}
+	pubkey := deposit.Data.PublicKey
+	index, ok := valIndexMap[bytesutil.ToBytes48(pubkey)]
```

Comment:
Shouldn't we do this check first? so if is `!ok` we dont have to `ProcessDeposit`

Reply:
But then no new deposit can ever be processed. Every new deposit will end up failing this check